_targets/build.project.armv7m7-imxrt106x

```diff
@@ -40,36 +40,99 @@ KERNEL_OFFS=$((0x11000))
 # Project specific build
 #
 
-PROGS=("dummyfs" "imxrt-multi" "psh" "imxrt-flash")
+PROGS=()
+BOOT_DEVICE="flash1"
+NAME_USER_SCRIPT="user.plo"
+OFFS_USER_SCRIPT=$((0x10a00))
+MAGIC_USER_SCRIPT=
+
+# Pre-init script is launched before user script
+PREINIT_SCRIPT=("map itcm 0x0 0x40000 rwx"
+                "map dtcm 0x20000000 0x20028000 rw"
+                "map ocram2 0x20200000 0x2027fe00 rwx"
+                "map xip1 0x70000000 0x70400000 rx"
+                "map xip2 0x60000000 0x64000000 rx"
+                "phfs usb0 1.2 phoenixd"
+                "phfs flash0 2.0 raw"
+                "phfs flash1 2.1 raw"
+                "console 0.1"
+                "wait 2000")
+
+# Obligatory applications to run Phoenix-RTOS
+USER_SCRIPT=("kernel ${BOOT_DEVICE}"
+             "app ${BOOT_DEVICE} -x dummyfs xip1 ocram2"
+             "app ${BOOT_DEVICE} -x imxrt-multi ocram2 ocram2"
+             "app ${BOOT_DEVICE} -x psh xip1 ocram2"
+             "app ${BOOT_DEVICE} imxrt-flash ocram2 ocram2")
+
+
+b_mkscript_user() {
+
+	if [ -f "$PREFIX_BUILD/plo/$NAME_USER_SCRIPT" ]; then
+		rm "$PREFIX_BUILD/plo/$NAME_USER_SCRIPT"
+	else
+		mkdir -p "$PREFIX_BUILD/plo/"
+	fi
 
-
-b_mkscript() {
 	ksz=$((($(wc -c < "${PREFIX_PROG_STRIPPED}phoenix-${TARGET_FAMILY}-${TARGET_SUBFAMILY}.elf") + SIZE_PAGE - 1) & PAGE_MASK))
 	poffs=$((KERNEL_OFFS + ksz))
 
-	for prog in "${PROGS[@]}"; do
-		psz=$((($(wc -c < "${PREFIX_PROG_STRIPPED}$prog") + SIZE_PAGE - 1) & PAGE_MASK))
-		printf "@%s(%x:%x)\n" "$prog" "$poffs" "$psz" >> "$PREFIX_BUILD/plo/script.plo"
-		((poffs+=psz))
-	done
-
-	cat _targets/armv7m7-imxrt106x.plo >> "$PREFIX_BUILD/plo/script.plo"
-	printf "\0" >> "$PREFIX_BUILD/plo/script.plo"
+	{
+		printf "%08x" "$MAGIC_USER_SCRIPT"
+		for cmd in "${USER_SCRIPT[@]}"; do
+			args=($cmd)
+			case "${args[0]}"
+			in
+				app)
+					if [ "${args[2]}" = "-x" ] ; then
+						name=$(cut -d ';' -f 1 <<< "${args[3]}")
+					else
+						name=$(cut -d ';' -f 1 <<< "${args[2]}")
+					fi
```

Comment:
```suggestion
					if [ "${args[2]}" = "-x" ] ; then
						name=$(${args[3]}%%;*)
					else
						name=$(${args[2]}%%;*)
					fi
```
It would be better to use parameter substitution.

Reply:
Yeah, shorter approach is better. Thanks.
However, you have made a small mistake, there should be `${args[3]%%;*}` instead of `${${args[3]}%%;*}`.